apps/web/src/components/docs/Docs.tsx

```diff
@@ -22,29 +24,69 @@ const getMDXComponent = mdxBundler.getMDXComponent;
 
 const DOCS_WRAPPER_ELEMENT_ID = 'embedded-docs';
 
+const globalOverrides = {
+  frameworkterminal: {
+    FrameworkTerminal: () => <nv-framework-terminal></nv-framework-terminal>,
+  },
+  framework: () => null,
+  echoterminal: {
+    EchoTerminal: () => <nv-framework-terminal></nv-framework-terminal>,
+  },
+};
+
 type DocsProps = PropsWithChildren<
   DocsQueryResults & {
     isLoading?: boolean;
-    actions: ReactNode;
+    actions?: ReactNode;
+    isChildDocs?: boolean;
+    mappings: Record<string, Record<string, string> | string>;
   }
 >;
 
 /*
  * Render the mdx for our mintlify docs inside of the web.
  * Fetching the compiled mdx from another service and then try to map the markdown to react components.
  */
-export const Docs = ({ code = '', description = '', title = '', isLoading, children, actions }: DocsProps) => {
+export const Docs = ({
+  code = '',
+  description = '',
+  title = '',
+  isLoading,
+  children: docsChildren,
+  actions = null,
+  isChildDocs = false,
+  mappings = {},
+}: DocsProps) => {
   const Component = useMemo(() => {
     if (code.length === 0) {
       return null;
     }
 
+    const globals = Object.keys(mappings).reduce((prev: Record<string, Record<string, any> | any>, key) => {
+      const path = mappings[key];
+
+      if (typeof path === 'string') {
+        prev[key] = () => <ChildDocs path={path} />;
+
+        return prev;
+      }
+
+      const value = path;
+
+      prev[key] = Object.keys(value).reduce((child: Record<string, any>, name) => {
+        child[name] = () => <ChildDocs path={child[name]} />;
+
+        return child;
+      }, {});
+
+      return prev;
+    }, {} as Record<string, Record<string, any> | any>);
+
     return getMDXComponent(code, {
-      frameworkterminal: {
-        FrameworkTerminal: () => <nv-framework-terminal></nv-framework-terminal>,
-      },
+      ...globals,
+      ...globalOverrides,
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
💭 **thought (non-blocking):** This is slightly confusing to me -- so we are always "overriding" a few specific properties? Would it be better to just switch the order of how these are spread so that they are used more as "defaults"?‏